Sources/TFMongoSwift/TensorGroup.swift

```diff
@@ -0,0 +1,323 @@
+            }
+        }
+    }
+    // swiftlint:enable cyclomatic_complexity
+
+    /// Returns a copy of this `TensorGroup` populated according to the given array of `TensorGroupMapping`s.
+    public func populated<M: TensorGroupMapping>(mappings: [M]) throws -> Self where M.Group == Self {
+        var copy = self
+        try copy.populate(mappings: mappings)
+        return copy
+    }
+
+    /// Returns a copy of this `TensorGroup` populated from the given MongoDB collection according to the provided
+    /// `TensorGroupMapping`.
+    public func populated<M: TensorGroupMapping & Codable>(uri: String? = nil,
+                                                           db: String,
+                                                           collection: String,
+                                                           filter: Document = [:],
+                                                           projection: Document? = nil,
+                                                           limit: Int64? = nil,
+                                                           mapping: M.Type) throws -> Self where M.Group == Self {
+        var copy = self
+        try copy.populate(uri: uri,
+                          db: db,
+                          collection: collection,
+                          filter: filter,
+                          projection: projection,
+                          limit: limit,
+                          mapping: M.self)
+        return copy
+    }
+
+    /// Populate this `TensorGroup` from the given MongoDB collection according to the provided `TensorGroupMapping`.
+    public mutating func populate<M: TensorGroupMapping & Codable>(uri: String? = nil,
+                                                                   db: String,
+                                                                   collection: String,
+                                                                   filter: Document = [:],
+                                                                   projection: Document? = nil,
+                                                                   limit: Int64? = nil,
+                                                                   mapping: M.Type) throws where M.Group == Self {
+        let client: MongoClient
+        if let uri = uri {
+            client = try MongoClient(uri)
+        } else {
+            client = try MongoClient()
+        }
+
+        let options = FindOptions(limit: limit, projection: projection)
+        let data = try Array(client.db(db).collection(collection, withType: M.self).find(filter, options: options))
+        try self.populate(mappings: data)
+    }
+
+    /// Returns a copy of this `TensorGroup` populated from the results of the given MongoDB aggregation query on the
+    /// given namespace according to the provided `TensorGroupMapping`.
+    /// `TensorGroupMapping`.
+    public func populated<M: TensorGroupMapping & Codable>(uri: String? = nil,
```

Comment:
Yeah, I included it to follow some patterns I saw in the `Dataset` TensorFlow api (e.g. `batched(batchSize:)`, `shuffled()`). These return copies instead of mutating the original.